File: akka-java/src/main/java/org/hahn/spider/actor/SaveWebSite.java
package org.hahn.spider.actor;

import akka.actor.UntypedActor;
import org.hahn.spider.message.Save;

/**
 * Created by jianghan on 16/5/2.
 */
public class SaveWebSite extends UntypedActor {

    @Override
    public void onReceive(Object msg) throws Exception {
        if (msg instanceof Save) {
            Save save = (Save) msg;
            System.out.println(save.toString());
        } else {
            unhandled(msg);
        }
    }

}

File: akka-scala/src/main/scala/hahn/learning/akka/spider/actor/SaveWebSite.scala
package hahn.learning.akka.spider.actor

import akka.actor.Actor
import hahn.learning.akka.spider.Message

/**
  * Created by jianghan on 16-5-26.
  */
class SaveWebSite extends Actor {

  def receive = {
    case Message.Save(url) => println(s"${url}")
    case _ => println("Error SaveWebSite")
  }

}
